src/kernels/jupyter/finder/remoteKernelFinder.ts

```diff
@@ -79,6 +79,8 @@ export class RemoteKernelFinder implements IRemoteKernelFinder, IDisposable {
         removed?: { id: string }[];
     }>();
     onDidChangeKernels = this._onDidChangeKernels.event;
+    private readonly _onDidChange = new EventEmitter<void>();
```

Comment:
You should be registering all events to the disposable, otherwise they could result in leaked memory when disposed.

Reply:
Already done, see Line 126, these are registered in the disposable,